src/main/java/org/victorrobotics/frc/dtlib/sensor/imu/DTIMU.java

```diff
@@ -0,0 +1,38 @@
+package org.victorrobotics.frc.dtlib.sensor.imu;
+
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+
+public interface DTIMU<IMUTYPE> extends Sendable, AutoCloseable {
```

Comment:
Can we make the interface not be parameterized on the concrete IMU type?

Reply:
This presents an issue with getting the internal object for any advanced usage, as it would have to return an `Object` and use an unchecked cast to some gyro class.

Reply:
For anyone who doesn't need the internal gyro object, the parameter can be omitted.

Reply:
Let's think about how client code will create and use these objects. There are two cases - generic and concrete.

Subsystem code might look like:
```
DTPigeonIMU imu = new DTPigeonIMU(...)
imu.internal().something()  // ok since it's known to be a DTPigeonIMU, and its internal() method returns a Pigeon2
```

Reusable utility code might look like:
```
public void doSomething(DTIMU imu) {
    // Can't meaningfully do anything that uses "internal()" because this code is supposed to work with all IMUs.
    // If it has different implementations using internal() for different IMUs, there would be 2 methods.
}
```

In either case, I don't think the `internal` method should be part of the interface, only of the concrete implementation class.

Reply:
The only reason I think `internal` needs to be a part of the interface is to make exposing advanced functionality a requirement; otherwise, you can never use a `DTTalonFX` object for any purpose other than its interface, so you’d need to switch to an ordinary `WPI_TalonFX`.

Reply:
We can implement internal() on both concrete classes without it being in the interface, since its return type depends on the specific concrete class.

Reply:
Okay, will do